src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -106,12 +112,11 @@ public int Count
             }
         }
 
+        public ref T this[int index] => ref _array[_offset + index];
```

Comment:
@jkotas I suggested it back in https://github.com/dotnet/corefx/issues/10528#issuecomment-265625210 and @KrzysztofCwalina approved of it. I thought it would be more flexible since you can do everything with a `ref T` (and more) that you can do with a regular indexer. If you say it is slower, however, I would be OK with having the latter.